control-plane/agents/src/bin/core/controller/reconciler/node/nexus.rs

```diff
@@ -0,0 +1,173 @@
+                Some(v) => {
+                    match v.operation_guard() {
+                        Ok(mut guard) => {
+                            let config = vol_spec.config().clone();
+                            if config.is_none() {
+                                tracing::error!("Failed to get config");
+                                continue;
+                            }
+                            let frontend_nodes = config.unwrap().frontend().node_names();
+                            if frontend_nodes.is_empty() {
+                                tracing::error!("Cannot republish, volume has no application");
+                                continue;
+                            }
+                            let frontend_node_id: NodeId = frontend_nodes[0].clone().into();
+                            tracing::info!(
+                                volume.uuid = vol_id.as_str(),
+                                nexus.uuid = nexus_id.as_str(),
+                                node.id = node_spec.id().as_str(),
+                                "Moving volume"
+                            );
+                            if let Err(e) =
+                                republish_volume(&mut guard, context, &vol_id, &frontend_node_id)
+                                    .await
+                            {
+                                tracing::error!(
+                                    error=%e,
+                                    volume.uuid = vol_id.as_str(),
+                                    nexus.uuid = nexus_id.as_str(),
+                                    node.id = node_spec.id().as_str(),
+                                    "Failed to republish volume"
+                                );
+                                continue;
+                            }
+                            volumes_moved += 1;
+                            tracing::info!(
+                                volume.uuid = vol_id.as_str(),
+                                nexus.uuid = nexus_id.as_str(),
+                                node.id = node_spec.id().as_str(),
+                                "Moved volume"
+                            );
+                        }
+                        Err(e) => {
+                            tracing::error!(error=%e, "Failed to get operation guard");
+                            continue;
+                        }
+                    };
+                }
+                None => {
+                    continue;
+                }
+            }
+        }
+    }
+    // Change the node state to "drained"
+    if volumes_moved == volumes_on_node {
+        if let Err(e) = context
+            .specs()
+            .set_node_drained(context.registry(), node_spec.id())
+            .await
+        {
+            tracing::error!(
+                error=%e,
+                node.id = node_id.as_str(),
+                "Failed to set node to state drained");
```

Comment:
formatting seems off here?

Reply:
Fixed. Not sure why the commit-checker didn't catch that.

Reply:
Yeah sadly rustfmt doesn't always work properly when macros are involved :(